### PR TITLE
feat: decode proxy transactions

### PR DESCRIPTION
**Pull request by Asmadek**
* support proxy pallet in call data decoder and extrinsic service 
* add tests for call data decoder and extrinsic service

### CODE REVIEW

File: src/renderer/entities/transaction/lib/__tests__/extrinsicService.test.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        maybeTimepoint: null,
+        otherSignatories: ['5F3cpq3CLZo77U6g6fHtSbpuhWfdXajs8DnBg9mBPBaPDtMB'],
+        threshold: 2,
+      },
+      chainId: '0x1234000000000000000000000000000000000000000000000000000000000000',
+      type: 'multisig_as_multi',
+    });
+  });
+
+  test.each([
```
Comment (egor0798) [inline src/renderer/entities/transaction/lib/__tests__/extrinsicService.test.ts:75]: `test.each`  does not provide great readability so I think you should move test data to separate file and create some type with meaningfull names. You can check `src/renderer/features/wallets/ImportKeys/lib/__tests__/import-keys-utils.test.ts` test there is a similar scenario
There is a type for test data
```
type ValidationTestData = {
  testName: string;
  derivation: DerivationWithPath;
  isValid: boolean;
};
```

and  that's how it used in test
```
test.each(importKeysMocks.shouldIgnoreDerivationTestData)('$testName', ({ derivation, shouldIgnore }) => {
      expect(importKeysUtils.shouldIgnoreDerivation(derivation)).toEqual(shouldIgnore);
    });
```

**Review by egor0798 — CHANGES_REQUESTED** [review]
Would be great if you improve test data format for `extrinsicService.test.ts`.
comment link:
https://github.com/novasamatech/nova-spektr/pull/1380/files#r1464908210

File: src/renderer/entities/transaction/lib/callDataDecoder.ts [hunk 2]
```diff
@@ -254,38 +283,59 @@ export const useCallDataDecoder = (): ICallDataDecoder => {
     [TransactionType.MULTISIG_AS_MULTI]: (decoded): Record<string, any> => {
       if (decoded.args.length === OLD_MULTISIG_ARGS_AMOUNT) {
         return {
-          threshold: decoded.args[0],
-          otherSignatories: decoded.args[1],
-          timepoint: decoded.args[2],
-          call: decoded.args[3],
-          storeCall: decoded.args[4],
-          maxWeight: decoded.args[5],
+          threshold: decoded.args[0].toString(),
```
[the remark this answers is not in the document]
Comment (tuul-wq) [reply src/renderer/entities/transaction/lib/callDataDecoder.ts:286]: By the way, why you appended `.toString`? Something was wrong without it?

File: src/renderer/shared/lib/utils/constants.ts [hunk 3]
```diff
@@ -20,6 +20,10 @@ export const SS58_PUBLIC_KEY_PREFIX = 1;
 export const TEST_ACCOUNT_ID = '0x08eb319467ea54784cd9edfbd03bbcc53f7a021ed8d9ed2ca97b6ae46b3f6014';
 export const TEST_ADDRESS = '1ChFWeNRLarAPRCTM3bfJmncJbSAbSS9yqjueWz7jX7iTVZ';
 export const TEST_SUBSTRATE_ADDRESS = '5CGQ7BPJZZKNirQgVhzbX9wdkgbnUHtJ5V7FkMXdZeVbXyr9';
+
+export const TEST_ACCOUNT_ID_2 = '0x83e0844510ede3aea6953c9886d9a51abdd944b6395de7b83bbce6dffce0c765';
```
Comment (tuul-wq) [inline src/renderer/shared/lib/utils/constants.ts:24]: Better to group with existing `TEST_ACCOUNT_ID`

**Review by tuul-wq — APPROVED** [review]

File: jest-setup.js [hunk 4]
```diff
@@ -15,6 +16,7 @@ global.ResizeObserver = jest.fn().mockImplementation(() => ({
 Object.defineProperty(global, 'crypto', {
   value: {
     randomUUID: () => '42',
+    getRandomValues: (arr) => crypto.getRandomValues(arr),
```
[the remark this answers is not in the document]
Comment (Asmadek) [reply jest-setup.js:19]: `MockProvider` requires it

File: src/renderer/entities/transaction/lib/callDataDecoder.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Asmadek) [reply src/renderer/entities/transaction/lib/callDataDecoder.ts]: We don't decode multisig txs before (only call data), and without transform it doesn't work

**Review by egor0798 — APPROVED** [review]

**Pull request merged**